### PR TITLE
server: fix wait conditions init after process restart

### PR DESCRIPTION
**Pull request by brig**

### CODE REVIEW

File: server/impl/src/main/java/com/walmartlabs/concord/server/process/waits/WaitProcessStatusListener.java [hunk 1]
```diff
@@ -52,9 +53,16 @@ public void onStatusChange(DSLContext tx, ProcessKey processKey, ProcessStatus s
     }
 
     private static void init(DSLContext tx, ProcessKey processKey) {
-        tx.insertInto(PROCESS_WAIT_CONDITIONS)
-                .columns(PROCESS_WAIT_CONDITIONS.INSTANCE_ID, PROCESS_WAIT_CONDITIONS.INSTANCE_CREATED_AT)
-                .values(processKey.getInstanceId(), processKey.getCreatedAt())
+        tx.insertInto(PROCESS_WAIT_CONDITIONS, PROCESS_WAIT_CONDITIONS.INSTANCE_ID, PROCESS_WAIT_CONDITIONS.INSTANCE_CREATED_AT)
```
Comment (ibodrov) [inline server/impl/src/main/java/com/walmartlabs/concord/server/process/waits/WaitProcessStatusListener.java:56]: Can we just do nothing here when `PREPARING`? If we skip the second insert anyway...

**Review by ibodrov — COMMENTED** [review]
The fix looks fine, but I've got a Q.

File: server/impl/src/main/java/com/walmartlabs/concord/server/process/waits/WaitProcessStatusListener.java (hunk 1 above)
Comment (brig) [reply server/impl/src/main/java/com/walmartlabs/concord/server/process/waits/WaitProcessStatusListener.java:56]: all root processes starts with NEW status, and forks do not enter this status; they go directly to PREPARING. I don't remember why it was done this way...

Comment (ibodrov) [reply server/impl/src/main/java/com/walmartlabs/concord/server/process/waits/WaitProcessStatusListener.java:56]: Ah cool. Then we maybe we should use two separate SQLs here, one with the existence check for PREPARING and the old one for NEW? Saves us an index read?

Comment (brig) [reply server/impl/src/main/java/com/walmartlabs/concord/server/process/waits/WaitProcessStatusListener.java:56]: user submits process: 
NEW -> PREPARING (just DB update, without history, or listeners notify)-> ENQUEUED -> ... -> 

restart process:
NEW -> PREPARING (just DB update, without history, or listeners notify)-> ENQUEUED -> ... -> 

child process:
NEW -> PREPARING (just DB update, without history, or listeners notify)-> ENQUEUED -> ... -> 

fork process:
PREPARING -> ENQUEUED -> ... ->

**Review by ibodrov — APPROVED** [review]

**Pull request merged**